rover_video/src/screenshot_server.hpp

```diff
@@ -0,0 +1,681 @@
+            cap = std::move(other.cap);  // Move cv ressources
+            video_writer = std::move(other.video_writer);
+            frame = std::move(other.frame);
+        }
+        return *this;
+    }
+
+    ~Recording()
+    {
+        if (cap.isOpened())  // avoid unnecessary logging when creating temporary objects
+        {
+            stopRecording.store(true);
+
+            if (recordingThread->joinable())
+            {
+                recordingThread->join();
+            }
+            // Release resources
+            this->cap.release();
+            this->video_writer.release();
+            this->appender.release();
+
+            RCLCPP_INFO(*logger_, "Recording stopped.");
+        }
+    }
+};
+
+class CameraNode : public rclcpp::Node
+{
+  public:
+  private:
+    rclcpp::Service<rover_msgs::srv::CameraControl>::SharedPtr _srv_control;
+    rclcpp::Subscription<rover_msgs::msg::GpsPosition>::SharedPtr _msg_position;
+    float last_latitude = 0.0, last_longitude = 0.0;
+
+    void controlIPCam(const rover_msgs::srv::CameraControl::Request& request, rover_msgs::srv::CameraControl::Response& response);
+    std::string getCurrentTime();
+    std::string getFileName(const std::string capture_name, std::string camURL, int state);
+    std::string getCamID(std::string cameraURL);
+    void callbackPosition(const rover_msgs::msg::GpsPosition& gps_message);
+    const std::string getFolderPath(int state);
+    bool folderExists(const std::string& path);
+    bool createFolder(const std::string& path);
+    bool getScreenshot(std::string screenshotFolderPath, std::string filename, std::string cameraURL);
+
+    bool newRecording(std::string videoFolderPath, std::string filename, std::string cameraURL);
+    bool stopRecording(std::string cameraURL);
+    bool StartWatchDog();
+    void VideoWatchDogFunction();
+    void RequestShutdown(std::string camURL);
+    std::atomic<bool> watchDogStop{false};
+    std::mutex recordingMutex;
+    std::thread videoWatchDog;
+    std::condition_variable recordingCv;
+
+    std::unordered_map<std::string, Recording> RecordingMap;
+    std::unordered_set<std::string> RecordingShutdownRequestSet;
+
+  public:
+    CameraNode();
+    ~CameraNode() {}
+};
+
+std::string CameraNode::getCamID(std::string cameraURL)
+{
+    std::string camID;
+
+    std::string::size_type nextDotPos;
+    std::string::size_type posID = cameraURL.find("144.");
+
+    if (posID != std::string::npos)
+    {
+        RCLCPP_DEBUG(LOGGER, "'144.' found.");
+        posID += 4;
+        nextDotPos = cameraURL.find(':', posID);
+
+        if (nextDotPos != std::string::npos)
+        {
+            camID = cameraURL.substr(posID, nextDotPos - posID);
+        }
+        else
+        {
+            camID = cameraURL.substr(posID);
+        }
+    }
+    else
+    {
+        RCLCPP_ERROR(LOGGER, "'144.' not found.");
+    }
+
+    RCLCPP_DEBUG(LOGGER, "Camera ID: %s", camID.c_str());
+
+    return camID;
+}
```

Comment:
ID's could be set in a std::map inside the rover_msgs package in the rovus_lib lib. This way all node needing this information could include this file from the lib:

for example:


```suggestion
std::map<size_t, std::string> CAMERA_NAME_MAP = { { 25, "Main camera" }, { 26, "Camera2" }, etc...};

```